src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class type) {
+        if (type.equals(Bulldozer.class)) {
+            return new ArrayList<>(bulldozerProducer.get());
```

Comment:
```suggestion
            return new ArrayList<>(new BulldozerProducer().get());
```

Reply:
But we'll create new SomeProducer every time when we call method getAll()? Why is it better?

Reply:
because you're creating them all 3 in every case, wasting some memory 
for example, if you need Bulldozer list, it's no need to create other objects